cmd/snap-update-ns/change.go

```diff
@@ -197,7 +210,8 @@ func changePerformImpl(c *Change, sec *Secure) (changes []*Change, err error) {
 		// As a result of this ensure call we may need to make the medium writable
 		// and that's why we may return more changes as a result of performing this
 		// one.
-		changes, err = c.ensureTarget(sec)
+		changesTarget, err = c.ensureTarget(sec)
+		changes = append(changes, changesTarget...)
 		if err != nil {
 			return changes, err
 		}
```

Comment:
It looks weird here and below that you append to changes before checking `err != nil` and then return those appended changes on error. Is this what you actually want? If so, can you add a comment as to why?

Reply:
Yes, we want to collect changes even if things fail. This is needed for undo.

Reply:
Can you add a comment to that effect then?